azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/vm/VirtualMachine.java

```diff
@@ -65,6 +66,36 @@ public Region getRegion() {
         return Region.fromName(remote().regionName());
     }
 
+    public boolean isSshEnabled() {
+        // TODO: @wangmi check if ssh is enabled, possible solution: INBOUND/TCP/22
+        // return remote().getPrimaryNetworkInterface().getNetworkSecurityGroup().securityRules().entrySet().stream()
+        //    .anyMatch(e -> SecurityRuleProtocol.TCP.equals(e.getValue().protocol()) &&
+        //        SecurityRuleDirection.INBOUND.equals(e.getValue().direction()) &&
+        //        "22".equals(e.getValue().destinationPortRange()) &&
+        //        "*".equals(e.getValue().destinationPortRange()));
+        return true;
+    }
+
+    public String getHostIp() {
+        return remote().getPrimaryPublicIPAddress().ipAddress();
+    }
+
+    public String getAdminUserName() {
+        return remote().innerModel().osProfile().adminUsername();
+    }
+
+    public String getAdminPassword() {
+        return remote().innerModel().osProfile().adminPassword();
+    }
+
+    public boolean isPasswordAuthenticationDisabled() {
+        final OSProfile profile = remote().innerModel().osProfile();
+        if (Objects.nonNull(profile.linuxConfiguration())) {
+            return profile.linuxConfiguration().disablePasswordAuthentication();
+        }
+        return true;
```

Comment:
return false here? As password should be the only authentication method for windows vm